### PR TITLE
Adds handlebars helpers

### PR DESCRIPTION
**Pull request by erindethorne**
Pretty much a straight port of the helpers we're using on members-site/projects-site. Anything that shouldn't be here? Anything missing?

### CODE REVIEW

File: plugins/handlebars.js [hunk 1]
```diff
@@ -15,3 +17,117 @@ app.Handlebars.registerHelper('applyPartial', function (templateName, vars, opti
...
+// notfirst helper.
+app.Handlebars.registerHelper('notfirst', function (context, options) {
+  if (options.data.index === 0) {
+    return options.inverse(this);
+  }
+  else {
+    return options.fn(this);
+  }
+});
+
```
Comment (cpsubrian) [inline plugins/handlebars.js:59]: Should we have 'last' and 'notlast'. Also .. camel case? Not sure on that.

File: plugins/handlebars.js [hunk 2]
```diff
@@ -15,3 +17,117 @@ app.Handlebars.registerHelper('applyPartial', function (templateName, vars, opti
...
+// else return plural
+// If forms are not passed, just return "s" for plural
+app.Handlebars.registerHelper('pluralize', function (array, singular, plural) {
+  if ('string' !== typeof singular) singular = '';
+  if ('string' !== typeof plural) plural = 's';
+  if (array.length && array.length === 1) {
+    return singular;
+  }
+  return plural;
+});
```
Comment (cpsubrian) [inline plugins/handlebars.js]: Could this support array.length as well as just a scalar value?

File: plugins/handlebars.js [hunk 3]
```diff
@@ -15,3 +17,117 @@ app.Handlebars.registerHelper('applyPartial', function (templateName, vars, opti
...
+
+app.Handlebars.registerHelper('or', function (var1, var2, options) {
+  if (var1 || var2) {
+    return options.fn(this);
+  }
+  else {
+    return options.inverse(this);
+  }
+});
```
Comment (cpsubrian) [inline plugins/handlebars.js]: I'd move this up with is/isnt  .. just organizationally.

Comment (cpsubrian) [inline plugins/handlebars.js]: s/this/these (or/and/nor)

File: plugins/handlebars.js (hunk 1 above)
Comment (erindethorne) [inline plugins/handlebars.js:59]: To be honest I don't know if I ever found a use case for `first` or `notfirst` - I almost didnt include them here. Not sure if adding the `last` and `notlast` would be useful or just more fluff. 

camelCase is fine with me.

File: plugins/handlebars.js (hunk 2 above)
Comment (erindethorne) [inline plugins/handlebars.js]: Yes. Good call.

File: plugins/handlebars.js [hunk 4]
```diff
@@ -15,3 +17,117 @@ app.Handlebars.registerHelper('applyPartial', function (templateName, vars, opti
...
+  // We use this when posting new comments to avoid the situation where the
+  // user's system clock is lagging the server's clock, which results in
+  // moment#fromNow() reports "in a few seconds", which is confusing and idiotic.
+  // Instead, if the server timestamp is "in the future", we just set it to the
+  // current time.
+  if (forceadjust && moment.utc(timestamp).diff() > 0) {
+    timestamp = moment.utc().format('YYYY-MM-DD HH:mm:ss');
+  }
+  return new app.Handlebars.SafeString('<span class="fromNow" data-timestamp="' + timestamp + '">' + moment.utc(timestamp).fromNow() + '</span>');
+});
```
Comment (cpsubrian) [inline plugins/handlebars.js]: Mixed feelings on cantina-web depending on moment. I actually think this should be part of the schema stuff (formatters?), so I wouldn't include these two.

Comment (erindethorne) [inline plugins/handlebars.js]: It would be a big convenience lost, but I guess I can get over it.

File: plugins/handlebars.js (hunk 1 above)
Comment (cpsubrian) [inline plugins/handlebars.js:59]: If we not using stuff lets not include it here. I'd rather keep cruft out unless its something we know we use a lot.

Comment (cpsubrian) [inline plugins/handlebars.js:59]: Because we are essentially inventing a handlebars DSL here .. so it should be minimal.

Comment (erindethorne) [inline plugins/handlebars.js:59]: :+1:

File: plugins/handlebars.js (hunk 3 above)
Comment (erindethorne) [inline plugins/handlebars.js]: :+1:

**Comment by erindethorne** [comment]
Better?

**Pull request merged**